### PR TITLE
Added external user api methods

### PR DESCRIPTION
**Pull request by rtorres87**

### CODE REVIEW

File: src/main/java/com/plenigo/sdk/internal/ApiResults.java [hunk 1]
```diff
@@ -219,4 +219,9 @@ private ApiResults() {
      */
     public static final String MOBILE_APP_SECRET = "mobileAppSecret";
 
+    /**
+     * the login token variable.
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/internal/ApiResults.java]: The upper case.

File: src/main/java/com/plenigo/sdk/internal/util/RestClient.java [hunk 2]
```diff
@@ -410,4 +414,22 @@ private void addHeadersToRequest(HttpURLConnection con, Map<String, String> head
...
         return call(apiUrl, DELETE_METHOD, resource, query, null, headers);
     }
+
+    /**
+     * This does a PUT HTTP call and accepts a body to be sent as a json object.
+     *
+     * @param apiUrl   The base API url
+     * @param resource The resource to call
+     * @param query    The query string to use
+     * @param body     The query string to use
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/internal/util/RestClient.java]: wrong comment

File: src/main/java/com/plenigo/sdk/internal/util/RestClient.java [hunk 3]
```diff
@@ -410,4 +414,22 @@ private void addHeadersToRequest(HttpURLConnection con, Map<String, String> head
...
+     * @param headers  the headers to send
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/internal/util/RestClient.java]: The upper case

File: src/main/java/com/plenigo/sdk/internal/util/RestClient.java [hunk 4]
```diff
@@ -410,4 +414,22 @@ private void addHeadersToRequest(HttpURLConnection con, Map<String, String> head
...
+     *
+     * @return The map result of the call
+     *
+     * @throws com.plenigo.sdk.PlenigoException whenever an error happens
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/internal/util/RestClient.java]: With namespace?

File: src/main/java/com/plenigo/sdk/internal/util/RestClient.java (hunk 2 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/internal/util/RestClient.java]: Fixed

File: src/main/java/com/plenigo/sdk/internal/ApiResults.java (hunk 1 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/internal/ApiResults.java]: What do you mean with the upper case

Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/internal/ApiResults.java]: Oh nevermind! got it!

File: src/main/java/com/plenigo/sdk/internal/util/RestClient.java (hunk 3 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/internal/util/RestClient.java]: Fixed

File: src/main/java/com/plenigo/sdk/internal/util/RestClient.java (hunk 4 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/internal/util/RestClient.java]: Fixed

**Pull request merged**